### PR TITLE
Move and Refactor convert to Template

### PR DESCRIPTION
**Pull request by WillCWX**
**What is the purpose of this pull request?**

- [ ] Documentation update
- [ ] Bug fix
- [ ] Feature addition or enhancement
- [X] Code maintenance
- [ ] DevOps
- [X] Improve developer experience
- [ ] Others, please explain:

Helps with #1756 
Helps make #2308  less messy

**Overview of changes:**

Moves `convert` from `Site/index.ts` to `Site/template.ts`.
Moves `readSiteConfig` from `Site/index.ts` to `SiteConfig.ts`.
Moves `_` to `constants.ts`

**Anything you'd like to highlight/discuss:**

This reduces Site/index.ts by 130 lines without affecting any functions.
Site/index.ts now has nothing to do with markbind init and markbind convert

template.ts is now more bloated but with the #2308  , it maybe made more streamline.

**Testing instructions:**

NIL other than `npm run test`

**Proposed commit message: (wrap lines at 72 characters)**

Move and refactor convert to template.ts

---

**Checklist:** :ballot_box_with_check:

- [ ] Updated the documentation for feature additions and enhancements
- [ ] Added tests for bug fixes or features
- [X] Linked all related issues
- [X] No unrelated changes

### CODE REVIEW

**Comment by lhw-1** [comment]
Hi @WillCWX, sorry for the delay & thanks for the PR! I'll get to reviewing it over the weekend.

File: packages/core/src/Site/template.ts [hunk 1]
```diff
@@ -1,17 +1,39 @@
...
+
+type NaviagablePage = {
+  src: string,
+  title?: string,
+};
 
 export class Template {
-  root: string;
+  rootPath: string;
   template: string;
```
Comment (lhw-1) [inline packages/core/src/Site/template.ts:30]: If we are changing `root` back to `rootPath`, then we may as well also change back `template` to `templatePath`?

**Review by lhw-1 — APPROVED** [review]
Thanks for the PR, @WillCWX! Sorry for the delay 😅

The PR seems to be a straightforward refactor and reshuffle of existing functionalities without changes to the logic, but helpful in the long run for refactoring `Site/index.js`.

File: packages/cli/src/cmd/init.js [hunk 2]
```diff
@@ -17,28 +16,25 @@ function init(root, options) {
   }
 
   const template = new Template(rootFolder, options.template);
-  template.init()
+  await template.init()
```
Comment (tlylt) [inline packages/cli/src/cmd/init.js:19]: If you `await` here, can we refactor the .then and .catch accordingly? (to stick to one way of handling asynchronous operations)
```
try {

} catch (error) {

}
```

**Comment by kaixin-hc** [comment]
Hi @WillCWX ! Took a look at this. You might not have noticed but codecov/patch is failing - I think the area that you are refactoring did not have tests. In this case since the overall project coverage didn't drop and a manual review of the code on my part looks okay, I'll merge this so that you can work on #2308 . But when you're over there, please check the tests for site.js and try to increase the coverage.

While these few lines might seem trivial, adding unit tests to cover them would help greatly especially with refactors like these, reducing the time needed for manual review and increasing our confidence in approving these changes.

**Pull request merged**